### PR TITLE
handle client closes watch connection

### PR DESCRIPTION
**Pull request by themanforfree**

### CODE REVIEW

**Review by rogercloud — COMMENTED** [review]
Please add tests to cover.

File: xline/tests/watch_test.rs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 10)]
+async fn test_watch_client_closes_connection() -> Result<(), Box<dyn Error>> {
+    let mut cluster = Cluster::new(3).await;
+    cluster.start().await;
+    let client = cluster.client().await;
+    let mut watch_client = client.watch_client();
+
+    let (watcher, mut stream) = watch_client.watch("foo", None).await?;
+    let handle = tokio::spawn(async move {
```
Comment (rogercloud) [inline xline/tests/watch_test.rs:51]: How to tell the correctness of tests? How do we know the server handles the client close operation correctly?

Comment (themanforfree) [reply xline/tests/watch_test.rs:51]: If we call this test before fixing it, the watch server will panic

File: xline/src/server/watch_server.rs [hunk 2]
```diff
@@ -48,25 +94,30 @@ struct WatchHandle {
     watcher_map: HashMap<WatchId, Watcher>,
     /// Next available `WatchId`
     next_id: WatchId,
+    /// Stop tx
+    stop_tx: mpsc::Sender<()>,
```
Comment (Phoenix500526) [inline xline/src/server/watch_server.rs:98]: If the messages transferred in the `stop` channel are meaningless, why don't we use a rendezvous channel here? Like `std::sync::mpsc`, `flume` supports 'rendezvous' channels, a bounded queue with a maximum capacity of zero will block senders until a receiver is available to take the message.

Comment (themanforfree) [reply xline/src/server/watch_server.rs]: done

**Review by rogercloud — APPROVED** [review]

**Pull request merged**